### DISCUSSION TITLE
interpolation导致的掉帧

### DISCUSSION THREAD

**Discussion by hong-tm** [hooke007/mpv_PlayKit#15] [Q&A 问答] [answered]
### 以下的视频绝大多数是从vcb下载，字幕实在论坛下载，希望可以有优化（我的电脑）的方案，这个interpolation，我的性能应该没什么问题吧？ 总结是，不开卡，开了可以缓解大部分卡但是还是有卡。。。hmm
![no interpolation overlord](https://user-images.githubusercontent.com/51999870/130665846-1d76674b-1fba-467c-9595-4e0e8a12ebd7.png) 
>在没有使用interpolation时候没有出现掉帧，但是镜头移动时候太快会一颤一颤

![interpolation overlord](https://user-images.githubusercontent.com/51999870/130666022-22d02ed8-8543-4ec1-a198-4ab12887d4a7.png)
![interpolation overlord2](https://user-images.githubusercontent.com/51999870/130666059-5c561a4a-877f-4f52-aa41-235b86915791.png) 
>使用后出现大量掉帧，镜头平滑但是掉帧印象观感，正常（内容播放没有特效字幕）时候镜头移动太快还是可以察觉视频整个在闪烁

![usage gpu interpolation overlord](https://user-images.githubusercontent.com/51999870/130666614-90f27b84-9a87-467f-a6d0-5b1e00ce712f.png)
>开启interpolation和Anime4K_Clamp_Highlights.glsl; Anime4K_Restore_CNN_Moderate_S.glsl; Anime4K_Upscale_CNN_x2_S.glsl后使用的gpu使用率

![interpolation nogame no live with ass some effect](https://user-images.githubusercontent.com/51999870/130667414-81995e95-79b5-41dc-82a5-d10666d47f4a.png)
>同样在一些行线（片尾）出现的特效会导致大量掉帧
[no game no life.zip](https://github.com/hooke007/MPV_lazy/files/7041382/no.game.no.life.zip)
>视频问题重现
[[UHA-WINGS][No Game No Life Zero][BDrip 1920x1080 x264 Flac].sc.zip](https://github.com/hooke007/MPV_lazy/files/7041260/UHA-WINGS.No.Game.No.Life.Zero.BDrip.1920x1080.x264.Flac.sc.zip)
游戏人生-字幕文件

![attack titan lag](https://user-images.githubusercontent.com/51999870/130667818-41f2f7fa-fb44-42d7-9f10-c4deb6545d90.png)
>之前因为字幕特效掉帧（片头）而关闭# video-sync = display-resample，这个是开启上面的滤镜后（之前说可能是因为CPU大核跑满问题）CPU显示图
[titan.zip](https://github.com/hooke007/MPV_lazy/files/7041427/titan.zip)
>巨人视频问题重现
[[LoliHouse] Shingeki no Kyojin - 39 [WebRip 1080p HEVC-yuv420p10 AAC ASSx2].SC.zip](https://github.com/hooke007/MPV_lazy/files/7041286/LoliHouse.Shingeki.no.Kyojin.-.39.WebRip.1080p.HEVC-yuv420p10.AAC.ASSx2.SC.zip)
>进击的巨人第三季-字幕文件

**Comment by hooke007** [comment, marked as answer]
Lolihouse的不少字幕都会带来这个问题，不是为了极致特效的话我不建议把它家的作为主力字幕（特别是电脑性能较弱的情况下）

1. 你是否看过 [这个](https://hooke007.github.io/mpv-lazy/[04]_%E5%B8%B8%E8%A7%81FAQ%EF%BC%88%E8%B8%A9%E5%9D%91%E5%BD%95%EF%BC%89.html#q-%E7%89%B9%E6%95%88%E5%AD%97%E5%B9%95%E9%83%A8%E5%88%86%E7%96%AF%E7%8B%82%E6%8E%89%E5%B8%A7%E5%85%B6%E4%BB%96%E9%83%A8%E5%88%86%E6%AD%A3%E5%B8%B8%E6%92%AD%E6%94%BE%EF%BC%89) ，禁用interpolation和我写的a1差不多，你可以先尝试a2

2. 另外笔记本双显卡渲染是不是存在问题我无法解答——只使用独显

3. 解码切换软解测试

按照顺序进行测试是否能降低负载和掉帧

=============

另外 Anime4K_Upscale_CNN_x2_S.glsl 没必要使用， 你的1080p屏不满足触发条件

=============

巨人这个字幕我有印象，之前有人报过这个问题，两个字幕都一样在我机器上只有个位数丢帧。最后我给的解决方案也是上边的a2

**Reply by hong-tm** [reply]
## 1.混合 （理想）
### 目前试了效果比较好的是这个设置（不使用Lolihouse字幕）
**全部都会出现播放瞬间掉几帧**应该是interpolation引起的，不知道有没有其他解决方法
````
hwdec              = no
video-sync                  = display-resample  
video-sync-max-video-change = 5           
interpolation               = yes            
tscale                      = oversample       
#blend-subtitles                = video
````
![Snipaste_2021-08-27_00-14-51](https://user-images.githubusercontent.com/51999870/130998510-0abc7323-74d8-43a9-b261-5a633dde0bfa.png)
>在mpv.conf里设置软解，在系统设置里面选择独显，开启interpolation和关闭字幕混合到视频帧上（#blend-subtitles = video）。

![on interpolation off blend-subtitles = video cpu and gpu setting](https://user-images.githubusercontent.com/51999870/130988311-61da055e-352b-4a0e-ae61-88a2fd771936.png)
>会使用部分CPU，独显和集显的资源，个位数掉帧，情况不错但是镜头平移还是看见晃
[on interpolation of blend best.zip](https://github.com/hooke007/MPV_lazy/files/7061163/on.interpolation.of.blend.best.zip)
目前最佳效果


## 2.纯软解（不理想）
![on interpolation off blend-subtitles = video cpu only some time laggy](https://user-images.githubusercontent.com/51999870/130988939-05263afc-1eb2-4cfd-916a-c412a121ad6b.png)
>使用纯软解和集显的状况，有时候不是很理想，主要还是有时性能不够，基本上可以播放但是镜头平移还是晃
---

## 后面都是纯硬解

## 3.开启interpolation和字幕混合到视频帧上增加了掉帧（不理想）
![on interpolation on blend-subtitles = video](https://user-images.githubusercontent.com/51999870/130989382-6eb60aee-9164-41f9-b374-4acb6a412996.png)
>纯硬解，开启blend-subtitles  = video之后大量掉帧，资源使用倒是没发现什么增加

## 4.开启interpolation和关闭字幕混合到视频帧上（不理想）
![on interpolation off blend-subtitles = video Radeon 540](https://user-images.githubusercontent.com/51999870/130993628-da21e604-e04c-4fe1-9ea4-e37b946ae481.png)
>之前一直使用的方式，部分情况掉帧，其余时间正常使用，镜头平移晃

## 5.关闭interpolation和开启字幕混合到视频帧上（不理想）
![off interpolation on blend-subtitles = video](https://user-images.githubusercontent.com/51999870/130994884-d90a8fa1-1141-45ab-8e94-e760302e655b.png)
>播放不掉帧但是镜头平移非常晃，感觉都分区在晃

## 6.关闭interpolation和字幕混合到视频帧上（不理想）
>播放跟上面一个一样

**Reply by hooke007** [reply]
这几个测试的时候有无对比着色器全部关闭的情况

**Reply by hong-tm** [reply]
没，明天测！

**Reply by hooke007** [reply]
忘了看字幕本身了。。15MB，特效有亿点点变态了，普通字幕都不会＞1MB

**Reply by hong-tm** [reply]
## 以下是在无滤镜下的结果，都没有掉帧或者只是个位数掉帧，但是晃动一直存在（某些特定段落），视频问题？
### 资源占用有断层因为我按到停止播放了！
**我其实也没太注意文件大小，主要看特效跟视频搭配就收了，每次lolihouse的好像比较好看**

---

![no filter of inte of blend](https://user-images.githubusercontent.com/51999870/131164195-382bc2c2-9ec8-44b5-84f2-abe1189fb18a.png)
>关闭interpolation，关闭字幕内嵌

![no filter of inte on blend](https://user-images.githubusercontent.com/51999870/131164398-7dd98551-e098-4dc7-9af2-7e6279c3966a.png)
>关闭interpolation，开启字幕内嵌，CPU占用增加

![no filter on inte of blend](https://user-images.githubusercontent.com/51999870/131164561-39b5aa84-d601-45f0-abf4-d5ba9134c71c.png)
>开启interpolation，关闭字幕内嵌

![no filter on inte on blend](https://user-images.githubusercontent.com/51999870/131164621-52e55503-329d-468e-b1d2-68f11c1addaf.png)
>开启interpolation，开启字幕内嵌，gpu增加一点点

---

[vlc test.zip](https://github.com/hooke007/MPV_lazy/files/7068604/vlc.test.zip)
>附上vlc的结果，一样会晃

![video com](https://user-images.githubusercontent.com/51999870/131169844-c2dae0a1-e9d3-423b-964c-588bb747731f.png)
>视频完整度应该没问题

**Reply by hong-tm** [reply]
还有一个事情，主要是播放视频是会突然的整个视频和声音都好像慢动作（视频声音一起慢），是我的RAM不太够的原因吗或者CPU问题？这个问题有时候突然出现蛮烦的

**Reply by hooke007** [reply]
这个晃动似乎只是judder而已，帧率和刷新率不匹配的正常现象。

慢动作我没有遇到过

无着色器不掉帧那说明性能实在捉襟见肘了....

**Reply by hong-tm** [reply]
这晃动是正常现象那就没什么办法了，毕竟就这笔记本性能不太行，上面那个混合选项还可以看，等之后再看看要不要换性能好一点的（设备）了，谢谢大佬帮助! 话说Anime4K_Restore_CNN_Moderate_M.glsl 有点难带 我都改Anime4K_Restore_CNN_Moderate_S.glsl 了，希望可以有轻量化一点的选择。

**Reply by hong-tm** [reply]
慢动作这个问题看之后加内存不知道能不能解决，每次都没录屏到这个问题，太突然了

**Reply by hooke007** [reply]
抗judder无非就三种方法：补帧，interpolation，刷新率匹配。
刷新率匹配是最低成本的做法，即把显示器刷新率调整到视频的（近似）整数倍。不过我个人觉得低于60hz的鼠标操作体验很糟糕

**Reply by hong-tm** [reply]
补帧，interpolation，刷新率匹配；interpolation倒是开了，有些效果但是还是不太完全，至于刷新率匹配我这边只有60hz啊，鼠标操作勉强能用毕竟没上过更高的没有对比

**Reply by hooke007** [reply]
我的意思是。。。降低到48/24hz，笔记本内屏可能需要第三方工具调节。interpolation的效果由tscale及其附属参数决定，你可以自行测试。

**Reply by natural-harmonia-gropius** [reply]
display fps那一行的第二个估值和第一个标称值差太多就说明性能不够确实掉帧了，慢动作也是性能不够导致desync触发resample。
理论上windows dwm会处理好与显示器同步的问题，性能不够导致mpv的窗口掉帧不如不开video-sync和interpolation

**Reply by natural-harmonia-gropius** [reply]
当然有gsync或者freesync就更好，监控的话用win+g的性能工具的fps栏好一点，cpu占用率意义不大

**Reply by hong-tm** [reply]
### 根据楼上的建议
![Snipaste_2021-09-02_23-39-43](https://user-images.githubusercontent.com/51999870/131886425-3929fa6f-826f-4f6c-a4b2-bb8351d01bfa.png)
![Snipaste_2021-09-03_00-00-05](https://user-images.githubusercontent.com/51999870/131886636-1ee85ee1-1cac-4fa8-8ebc-f06f5360f86a.png)
>在interpolation和开启滤镜下

![Snipaste_2021-09-02_23-56-55](https://user-images.githubusercontent.com/51999870/131886771-d2769297-c6df-4d0d-95b4-caaf8d6b489c.png)
>关闭interinterpolation和所有滤镜

![Snipaste_2021-09-02_23-43-29](https://user-images.githubusercontent.com/51999870/131886841-f0cedafd-2881-4fd1-8e8f-4d7a1b2e51f8.png)
![Snipaste_2021-09-02_23-46-33](https://user-images.githubusercontent.com/51999870/131887116-6a42a561-78b4-4c26-8c49-47a659017980.png)
>使用Custom Resolution Utility后开启FreeSync，会出现断断续续黑屏（放弃），之前修改刷新率也会出现屏幕炸裂的情况

**Reply by hong-tm** [reply]
> 我的意思是。。。降低到48/24hz，笔记本内屏可能需要第三方工具调节。interpolation的效果由tscale及其附属参数决定，你可以自行测试。

尝试更改tscale= oversample去另外两个都会出现掉帧，prof.conf的配置也改过会出现掉帧

**Reply by natural-harmonia-gropius** [reply]
freesync强开没必要。总的来说还是性能问题，开硬解或者移除几个着色器保证帧率为先，毕竟细节比卡顿难察觉的多

**Reply by hooke007** [reply]
掉帧是性能不足已经确定了。。。我那里讲的是另外的两件事
已经卡的情况下改tscale没用处，，我的意思是不卡的情况改tscale观察画面顺滑度去选一个最适合你的
而且。。tscale也只有开启interpolation的时候才有用。

**Reply by hong-tm** [reply]
好的，谢谢帮助！
